### PR TITLE
Preview map only on click in user’s dashboard

### PR DESCRIPTION
**Pull request by davidbgk**
Refs #1459

This is a proposal to reduce the size/time for that view. The map is only loaded when the user clicks the Preview button.

Before:

Capture d’écran, le 2023-12-19 à 09 34 20

After:
Capture d’écran, le 2023-12-19 à 14 20 08

The preview opens in an almost full-screen dialog element with blurred background:

Capture d’écran, le 2023-12-19 à 14 24 06

Let's discuss :) Once accepted, I'll put styles and scripts to their regular places.

### CODE REVIEW

File: umap/models.py [hunk 1]
```diff
@@ -193,6 +194,40 @@ class Map(NamedModel):
     objects = models.Manager()
     public = PublicManager()
 
+    @property
+    def unique_id(self):
+        return f"map_{self.pk}"
+
+    @property
+    def preview_settings(self):
+        from .views import _urls_for_js
```
Comment (yohanboniface) [inline umap/models.py:203]: Move to some helper file ?

File: umap/models.py [hunk 2]
```diff
@@ -193,6 +194,40 @@ class Map(NamedModel):
     objects = models.Manager()
     public = PublicManager()
 
+    @property
+    def unique_id(self):
+        return f"map_{self.pk}"
```
Comment (yohanboniface) [inline umap/models.py:199]: This sounds like more related to HTML than models, shouldn't we keep `map_` inside the html and just use `pk` ?

**Review by yohanboniface — COMMENTED** [review]
Sounds good!  :+1:

**Comment by davidbgk** [comment]
Current status:

Capture d’écran, le 2023-12-27 à 13 30 51

**Pull request merged**